api/lib/application/users/index.js

```diff
@@ -8,8 +8,42 @@ const { passwordValidationPattern } = require('../../config').account;
 const { EntityValidationError } = require('../../domain/errors');
 const identifiersType = require('../../domain/types/identifiers-type');
 
+const adminRoutes = [
+  // GET
+  {
+    method: 'GET',
+    path: '/api/admin/users',
+    config: {
+      pre: [
+        {
+          method: (request, h) =>
+            securityPreHandlers.userHasAtLeastOneAccessOf([
+              securityPreHandlers.checkUserHasRoleSuperAdmin,
+              securityPreHandlers.checkUserHasRoleCertif,
+              securityPreHandlers.checkUserHasRoleSupport,
+              securityPreHandlers.checkUserHasRoleMetier,
+            ])(request, h),
+          assign: 'hasAuthorizationToAccessAdminScope',
```

Comment:
avant on avait :
```
   {
            method: securityPreHandlers.checkUserHasRoleSuperAdmin,
            assign: 'hasAuthorizationToAccessAdminScope',
          },
```

je pense qu'on peut attribuer une nouvelle clé pour l'utilisation de `userHasAtLeastOneAccessOf`

Je comprends pas bien l'intérêt de `assign` (la doc ne m'a pas trop éclairée https://hapi.dev/api/?v=20.2.2#-routeoptionspre)  ici mais vu que c'est une clé, peut-être que ça serait mieux de lui en spécifier une nouvelle

Reply:
Comme la documentation le décris : assign - key name used to assign the response of the method to in `request.pre` and `request.preResponses`, cela permettrait la récupération de la valuer de retour de la méthode du pre-handler dans le.s pre-handler.s suivant et également au niveau du controller du moment que l'on est accès à la variable `request` de Hapi.

Reply:
C'était plus, je comprends pas ce que fait cet assign dans notre code.
je l'ai supprimé dans la route pour récupérer les users `/api/admin/users` et j'ai pas vu d'effet.

Reply:
Dans le cadre cette PR et je pense même pour beaucoup d'autres routes, le assign n'est pas utile. Dans la 1ère réflexion, je comptais ne pas l'utiliser mais j'ai changé d'avis et essayé de garder le même code que toutes les routes ayant ce nom de propriété.